lib/backend/k8s/resources/node.go

```diff
@@ -306,17 +306,19 @@ func K8sNodeToCalico(k8sNode *kapiv1.Node, usePodCIDR bool) (*model.KVPair, erro
 func fillAllAddresses(calicoNode *apiv3.Node, k8sNode *kapiv1.Node) {
 	if bgp := calicoNode.Spec.BGP; bgp != nil {
 		if addr := bgp.IPv4Address; addr != "" {
-			calicoNode.Spec.Addresses = append(calicoNode.Spec.Addresses, apiv3.NodeAddress{Address: addr})
+			calicoNode.Spec.Addresses = append(calicoNode.Spec.Addresses, apiv3.NodeAddress{Address: addr, Type: "bgp"})
```

Comment:
Types should follow Kubernetes API conventions for enums.

So basically casing.

```
- BGP
- ExternalIP
- InternalIP
```

IMO there's no reason to reinvent the types. Let's just use the same values k8s already does.

Reply:
The "BGP" value is probably not what we want to call it, given it's used for more than just BGP nowadays anyway, and isn't orthogonal with InternalIP (most cases they are probably the same thing). For example, this IP will be set in VXLAN mode even with BGP/BIRD disabled entirely. 

I would suggest one of the following:
- Treat this address as an `InternalIP` and have the felix code use the first InternalIP it finds in the list.
- Call this type something more generic - "AutodetectedIP" or something.

Reply:
`CalicoNodeIP` ?

Reply:
CalicoNodeIP is a bit ambiguous as well isn't it? Because calico/node might be using an IP that isn't a CalicoNodeIP after this PR is merged :grin: 

I'd rather not have APIs tied to a particular software component (it's contribued, but what if we decide in the future that some component other than calico/node should do IP auto-detection?)

Reply:
I was thinking of `CalicoNodeIP` as the "Node IP that has been configured in the Calico datamodel" rather than the "IP determined by calico/node" but `AutodetectedIP` or `CalicoAutodetectedIP` would be fine by me.  @sridhartigera suggest you and @caseydavenport discuss this and come to an agreement.

Reply:
Works for me!

Reply:
Can we go with AutoDetectedIP?

Reply:
I'm fine with CalicoNodeIP for now